ganga/GangaCore/Lib/Localhost/LocalHostExec.py.template

```diff
@@ -85,6 +88,27 @@ errorfile=open('stderr','w')
 sys.stdout=open('./__syslog__','w')
 sys.stderr=sys.stdout
 
+from Virtualization import checkDocker, checkUDocker, checkSingularity, installUdocker
```

Comment:
I think we need to be careful here. Is it such a good idea to be importing things from ganga? This will cause issues with different python versions.

Reply:
I'm sending the virtualization.py module to the backend. so it is not calling a gangaclass it's calling a class within the same folder

Reply:
I think @mesmith75 point was about versions. e.g. if we switch the ganga code to Python 3 will this still work

Reply:
Virtualization module will only be used in backend. So I think we don't need to upgrade the virtualization module to python 3. We can maintain it as python 2.7.  But it will be bit odd to have only one class in 2.7 within the ganga code base.

Reply:
Let us discuss this issue at the meeting today. While we have agreed that the main part of Ganga very soon will transfer to python3, then the backend template scripts might need to be kept such that they are both python2 and python3 compatible.

Reply:
I had a brief look at your `Virtualization` utility class and it looks python 2/3 compatibly already so it may well be that this is a non-issue. Could you just try running the 2to3 tool over this module to check.

```bash
pip install 2to3
2to3 Virtualization.py
```